src/cli/crd/generate.ts

```diff
@@ -0,0 +1,316 @@
+
+import { Command } from "commander";
+import fs from "fs";
+import path from "path";
+import { stringify as toYAML } from "yaml";
+import {
+  Project,
+  InterfaceDeclaration,
+  TypeAliasDeclaration,
+  SyntaxKind,
+  Node,
+  SourceFile,
+  Type,
+} from "ts-morph";
+import { createDirectoryIfNotExists } from "../../lib/filesystemService";
+import { kind as k } from "kubernetes-fluent-client";
+import { V1JSONSchemaProps } from "@kubernetes/client-node";
+
+export default new Command("generate")
+  .description("Generate CRD manifests from TypeScript definitions")
+  .option("--output <output>", "Output directory for generated CRDs", "./crds")
+  .action(generateCRDs);
+
+export function extractCRDDetails(
+  content: string,
+  sourceFile: SourceFile,
+): {
+  kind: string | undefined;
+  fqdn: string;
+  scope: "Cluster" | "Namespaced";
+  plural: string;
+  shortNames?: string[];
+} {
+  const kind = extractSingleLineComment(content, "Kind");
+  const group = extractSingleLineComment(content, "Group") ?? "example";
+  const domain = extractSingleLineComment(content, "Domain") ?? "pepr.dev";
+  const details = extractDetails(sourceFile);
+
+  const fqdn = `${group}.${domain}`;
+
+  const { plural, scope } = details;
+
+  const shortNames = details.shortName ? [details.shortName] : undefined;
+  return { kind, plural, scope, shortNames, fqdn };
+}
+
+export async function generateCRDs(options: { output: string }): Promise<void> {
+  const outputDir = path.resolve(options.output);
+  await createDirectoryIfNotExists(outputDir);
+
+  const project = new Project();
+  const apiRoot = path.resolve("api");
+  const versions = getAPIVersions(apiRoot);
+
+  for (const version of versions) {
+    const sourceFiles = loadVersionFiles(project, path.join(apiRoot, version));
+    for (const sourceFile of sourceFiles) {
+      processSourceFile(sourceFile, version, outputDir);
+    }
+  }
+}
+
+export function getAPIVersions(apiRoot: string): string[] {
+  return fs.readdirSync(apiRoot).filter(v => fs.statSync(path.join(apiRoot, v)).isDirectory());
+}
+
+export function loadVersionFiles(project: Project, versionDir: string): SourceFile[] {
+  const files = fs.readdirSync(versionDir).filter(f => f.endsWith(".ts"));
+  const filePaths = files.map(f => path.join(versionDir, f));
+  return project.addSourceFilesAtPaths(filePaths);
+}
+
+export function processSourceFile(
+  sourceFile: SourceFile,
+  version: string,
+  outputDir: string,
+): void {
+  const content = sourceFile.getFullText();
+  const { kind, fqdn, scope, plural, shortNames } = extractCRDDetails(content, sourceFile);
+
+  if (!kind) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing '// Kind: <KindName>' comment`);
+    return;
+  }
+
+  const spec = sourceFile.getInterface(`${kind}Spec`);
+  if (!spec) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing interface ${kind}Spec`);
+    return;
+  }
```

Comment:
```suggestion
  if (!kind) {
    console.warn(`Skipping ${sourceFile.getBaseName()}: missing '// Kind: <KindName>' comment`);
    return;
  }

  const spec = sourceFile.getInterface(`${kind}Spec`);
  if (!spec) {
    console.warn(`Skipping ${sourceFile.getBaseName()}: missing interface ${kind}Spec`);
    return;
  }
```

I think the use of emoji in Pepr log messages is inconsistent with the rest of the project's style? I only see emoji in some `console.info()` messages in the CLI and that's it.